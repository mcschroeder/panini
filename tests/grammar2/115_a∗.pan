import axioms

f115 : {s:string|?} -> unit
f115 = \s0:string.  
  rec w : string -> int -> unit = \s:string. \i:int.
    let n = length s in
    let p = lt i n in
    if p then
      let x = charAt s i in
      let p2 = match x "a" in
      let _ = assert p2 in
      let i2 = add i 1 in
      w s i2
    else
      unit
  in
    w s0 0
