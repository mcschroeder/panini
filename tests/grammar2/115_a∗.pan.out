f115 : {s:𝕊 | s ∈ a*} → 𝟙